external/upstream/cppe/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+if(${ENABLE_cppe})
+    find_package(cppe 0.0.8 CONFIG QUIET)
+
+    if(${cppe_FOUND})
+        get_property(_loc TARGET cppe::cppe PROPERTY LOCATION)
+        message(STATUS "${Cyan}Found cppe${ColourReset}: ${_loc} (found version ${cppe_VERSION})")
+        add_library(cppe_external INTERFACE)  # dummy
+    else()
+        if(${CMAKE_INSIST_FIND_PACKAGE_cppe})
+            message(FATAL_ERROR "Suitable cppe could not be externally located as user insists")
+        endif()
+
+        include(ExternalProject)
+        message(STATUS "Suitable cppe could not be located, ${Magenta}Building cppe${ColourReset} instead.")
+        ExternalProject_Add(cppe_external
+            # GIT_REPOSITORY https://github.com/maxscheurer/cppe
+            # GIT_TAG master
+            URL https://github.com/maxscheurer/cppe/archive/v0.0.8.tar.gz
+            # UPDATE_COMMAND ""
+            CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${STAGED_INSTALL_PREFIX}
+                       -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                       -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                       -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                       -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
+                       -DCMAKE_INSTALL_BINDIR=${CMAKE_INSTALL_BINDIR}
+                       -DCMAKE_INSTALL_DATADIR=${CMAKE_INSTALL_DATADIR}
+                       -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}
+                       -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
+                       -DENABLE_OPENMP=${ENABLE_OPENMP}
```

Comment:
How much of a performance hit would turning off OpenMP (in cppe, not psi4) be? Not permanently but as a temporary, let's-get-this-integrated measure.

I suggest this because https://github.com/maxscheurer/cppe/blob/master/cppe/CMakeLists.txt#L12-L18 is fine for standalone, but it's going to choose a threading library based only on compiler, not taking LAPACK threading or `OpenMP_LIBRARY_DIRS` into account. Psi uses the crazy [`MathOpenMP`](https://github.com/psi4/psi4/blob/master/external/common/lapack/FindMathOpenMP.cmake) to get this right, a scheme I've no affection for but guardedly trust. We can find a way to hand the correct openmp flags to cppe (which is how we do lapack for other addons), but I don't want that detail to hold up this PR.

Reply:
Won't be an issue I guess. One can always provide a self-built `cppe` via `PYTHONPATH` that is built with OpenMP for the time being.
Will turn it off.

Reply:
Done.